### PR TITLE
add metrics endoints

### LINKED ISSUE #4175: Create metric endpoint for node deployments
**User Story**
Use `kubernetes/metrics` to retrive metrics for node delpoyments:
GET /api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments/{nodedeployment_id}/metrics

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add a new endpoint for node deployment metrics. Metrics are associated with nodes by name:
```
[{"name":"worker-gxgng-67cdd57c98-wpkdx","memory":511,"memoryUsage":74,"cpu":209,"cpuUsage":26},{"name":"worker-gxgng-67cdd57c98-2xzz5","memory":503,"memoryUsage":73,"cpu":100,"cpuUsage":12}]
```
Memory in bytes, cpu in mili cores, usage in percentage

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4175

**Special notes for your reviewer**:

```release-note
add a new endpoint for node deployment metrics:
GET /api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments/{nodedeployment_id}/metrics
```

### CODE REVIEW

**Comment by zreigz** [comment]
/hold
I have to fix and add unit tests

File: api/pkg/cluster/client/client.go [hunk 1]
```diff
@@ -121,3 +123,37 @@ func (p *provider) GetClient(c *kubermaticv1.Cluster, options ...ConfigOption) (
 
 	return p.restMapperCache.Client(config)
 }
+
+func (p *provider) GetMetricsClient(c *kubermaticv1.Cluster) (metricsclient.MetricsV1beta1Interface, error) {
```
Comment (alvaroaleman) [inline api/pkg/cluster/client/client.go]: What is the reason for not using the dynamic client for this?

Comment (zreigz) [reply api/pkg/cluster/client/client.go]: When I took ready to use library I didn't think about the dynamic client. I've checked it and works with the dynamic client. Thanks

File: api/pkg/api/v1/types.go [hunk 2]
```diff
@@ -1018,6 +1018,20 @@ type ClusterMetric struct {
 	Values []float64 `json:"values,omitempty"`
 }
 
+// NodeMetric defines a metric for the given node
+// swagger:model NodeMetric
+type NodeMetric struct {
+	Name string `json:"name"`
+	// Memory in bytes
+	Memory int64 `json:"memory,omitempty"`
```
Comment (alvaroaleman) [inline api/pkg/api/v1/types.go]: Nit: What about renaming the fields to:
* `MemoryTotalBytes`
* `MemoryUsedPercentage`
* `CPUTotalMillicores`
* `CPUUsedPercentage`

? That way ppl can understand the fields without having to look at the docs, right now its IMHO not clear from the name what exactly is in there and what unit is used

**Comment by zreigz** [comment]
@alvaroaleman PTAL

**Review by alvaroaleman — APPROVED** [review]
/lgtm
/approve

**Pull request merged**